daemon/id.go

```diff
@@ -0,0 +1,30 @@
+package daemon // import "github.com/docker/docker/daemon"
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/docker/docker/pkg/ioutils"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+func loadOrCreateID(idPath string) (string, error) {
+	err := os.MkdirAll(filepath.Dir(idPath), 0755)
```

Comment:
FWIW; I was looking at splitting out some of these changes, and I think this is wrong; `idPath` is `filepath.Join(config.Root, "engine_id")`, which means that this function could potentially set up the daemon's root directory with `0755` permissions.